tests/core/test_db.py

```diff
@@ -21,47 +35,76 @@ def test_check_database():
 
 
 @pytest.mark.parametrize(
-    "patch_settings",
-    [{"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": True}],
+    ("patch_settings", "caplog"),
+    [
+        ({"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": True}, None),
+        ({"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": False}, None),
+        ({"ENVIRONMENT": "production"}, None),
+    ],
     indirect=True,
 )
-def test_init_database_none(patch_settings: Settings):
-    org_exec = Session.exec
-    Session.exec = MagicMock()
-    Session.exec.return_value.first.return_value = None
-
+def test_init_database(patch_settings: Settings, caplog: pytest.LogCaptureFixture):
     init_db()
-
-    expected = [
-        (select(User).where(User.name == "Robbert"),),
-        (select(Status).where(Status.name == "Todo"),),
-        (select(Task).where(Task.title == "First task"),),
-    ]
-
-    for i, call_args in enumerate(Session.exec.call_args_list):
-        assert str(expected[i][0]) == str(call_args.args[0])
-
-    Session.exec = org_exec
-
-
-@pytest.mark.parametrize(
-    "patch_settings",
-    [{"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": True}],
-    indirect=True,
-)
-def test_init_database(patch_settings: Settings):
-    org_exec = Session.exec
-    Session.exec = MagicMock()
-
-    init_db()
-
-    expected = [
-        (select(User).where(User.name == "Robbert"),),
-        (select(Status).where(Status.name == "Todo"),),
-        (select(Task).where(Task.title == "First task"),),
-    ]
-
-    for i, call_args in enumerate(Session.exec.call_args_list):
-        assert str(expected[i][0]) == str(call_args.args[0])
-
-    Session.exec = org_exec
+    if patch_settings.ENVIRONMENT == "demo":
```

Comment:
Why do we now check the logs instead of the actual functions called? I prefer the actual functions because then you check what is actually happening instead of the logs which is not the main logic of the function.

Reply:
Valid point. I did not know a better way to check if a function had been called, unless using a Mock which I prefer not to use because then I don't know if the logic works (only that the function has been called). I did however find a 'count calls' decorater, I will see if I can use that.

Reply:
Maybe you can help @berrydenhartog as I have been using a wrapper around a function to be able to count calls, but in this case need to change an existing function and I can't get that to work.